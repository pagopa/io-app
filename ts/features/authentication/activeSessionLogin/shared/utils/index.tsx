import I18n from "i18next";
import { Millisecond } from "@pagopa/ts-commons/lib/units";
import { Platform } from "react-native";
import { CieAuthenticationErrorReason } from "../../../login/cie/store/actions/cie";
import { ReadingState } from "../../../login/cie/components/CieCardReadingAnimation";

// the timeout we sleep until move to consent form screen when authentication goes well
export const WAIT_TIMEOUT_NAVIGATION = 1700 as Millisecond;
export const WAIT_TIMEOUT_NAVIGATION_ACCESSIBILITY = 5000 as Millisecond;
export const VIBRATION = 100 as Millisecond;
export const accessibityTimeout = 100 as Millisecond;

// A subset of Cie Events (errors) which is of interest to analytics
export const analyticActions = new Map<CieAuthenticationErrorReason, string>([
  // Reading interrupted before the sdk complete the reading
  ["Transmission Error", I18n.t("authentication.cie.card.error.onTagLost")],
  ["ON_TAG_LOST", I18n.t("authentication.cie.card.error.onTagLost")],
  [
    "TAG_ERROR_NFC_NOT_SUPPORTED",
    I18n.t("authentication.cie.card.error.unknownCardContent")
  ],
  [
    "ON_TAG_DISCOVERED_NOT_CIE",
    I18n.t("authentication.cie.card.error.unknownCardContent")
  ],
  ["PIN Locked", I18n.t("authentication.cie.card.error.generic")],
  ["ON_CARD_PIN_LOCKED", I18n.t("authentication.cie.card.error.generic")],
  ["ON_PIN_ERROR", I18n.t("authentication.cie.card.error.tryAgain")],
  ["PIN_INPUT_ERROR", ""],
  ["CERTIFICATE_EXPIRED", I18n.t("authentication.cie.card.error.generic")],
  ["CERTIFICATE_REVOKED", I18n.t("authentication.cie.card.error.generic")],
  ["AUTHENTICATION_ERROR", I18n.t("authentication.cie.card.error.generic")],
  [
    "EXTENDED_APDU_NOT_SUPPORTED",
    I18n.t("authentication.cie.nfc.apduNotSupported")
  ],
  [
    "ON_NO_INTERNET_CONNECTION",
    I18n.t("authentication.cie.card.error.tryAgain")
  ],
  ["STOP_NFC_ERROR", ""],
  ["START_NFC_ERROR", ""]
]);

export type TextForState = {
  title: string;
  subtitle?: string;
  content: string;
};

// some texts changes depending on current running Platform
export const getTextForState = (
  state: ReadingState.waiting_card | ReadingState.error,
  errorMessage: string = ""
): TextForState => {
  const texts: Record<
    ReadingState.waiting_card | ReadingState.error,
    TextForState
  > = Platform.select({
    ios: {
      [ReadingState.waiting_card]: {
        title: I18n.t("authentication.cie.card.titleiOS"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeaderiOS"),
        // the native alert hides the screen content and shows a message it self
        content: ""
      },
      [ReadingState.error]: {
        title: I18n.t("authentication.cie.card.error.readerCardLostTitle"),
        subtitle: "",
        // the native alert hides the screen content and shows a message it self
        content: ""
      },
      [ReadingState.reading]: {
        title: I18n.t("authentication.cie.card.titleiOS"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeaderiOS"),
        // the native alert hides the screen content and shows a message it self
        content: ""
      }
    },
    default: {
      [ReadingState.waiting_card]: {
        title: I18n.t("authentication.cie.card.title"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeader"),
        content: I18n.t("authentication.cie.card.layCardMessageFooter")
      },
      [ReadingState.error]: {
        title: I18n.t("authentication.cie.card.error.readerCardLostTitle"),
        subtitle: I18n.t("authentication.cie.card.error.onTagLost"),
        content: errorMessage
      }
    }
  });
  return texts[state];
};
