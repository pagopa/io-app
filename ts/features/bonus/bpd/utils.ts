import { PatchedWalletV2 } from "../../../types/pagopa";

/**
 * Return true if bpd is enabled for the wallet (aka payment method)
 * TODO: replace string with enum
 * @param wallet
 */
export const hasBpdCapability = (wallet: PatchedWalletV2) =>
  wallet.enableableFunctions.includes("BPD");
