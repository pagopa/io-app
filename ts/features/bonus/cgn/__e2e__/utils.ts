import fetch from "node-fetch";
import { e2eWaitRenderTimeout } from "../../../../__e2e__/config";
import I18n from "../../../../i18n";

export const activateCGNBonusSuccess = async () => {
  const startActivationCta = element(by.id("activate-bonus-button"));
  await waitFor(startActivationCta)
    .toBeVisible()
    .withTimeout(e2eWaitRenderTimeout);
  await startActivationCta.tap();
  await waitFor(element(by.id("cgnConfirmButtonTestId")))
    .toBeVisible()
    .withTimeout(e2eWaitRenderTimeout);

  // We should unsubscribe after the activation, in order to allow the next test run (we can remove this part when we will have the possibility
  // to reset the dev-server with an API command.

  // Go to bonus details
  await element(by.id("cgnConfirmButtonTestId")).tap();

  // wait for unsubscribe cta
  const scrollView = element(by.id("CGNCardDetailsScrollView"));

  // The section has a loading spinner on top of
  // everything so we must wait for it to disappear
  await waitFor(scrollView)
    .toBeVisible()
    .withTimeout(2 * e2eWaitRenderTimeout);

  // make sure to scroll to bottom, otherwise in small devices the element will not be visible nor tappable
  await scrollView.scrollTo("bottom");

  const unsubscribeCgnCta = element(by.id("cgnDeactivateBonusTestId"));
  await waitFor(unsubscribeCgnCta)
    .toBeVisible()
    .withTimeout(e2eWaitRenderTimeout);
  // unsubscribe
  await unsubscribeCgnCta.tap();

  // confirm alert
  const alertCTA = element(by.text(I18n.t("global.buttons.deactivate")));
  await waitFor(alertCTA).toBeVisible().withTimeout(e2eWaitRenderTimeout);
  await alertCTA.tap();
};

export const deactivateCGNCardIfNeeded = async () => {
  // This is needed since an E2E can fail at any moment. If it does so
  // after the card has been activated, all subsequent retries and tests
  // will fail, since the card is already activated. Be aware that this
  // endpoint is not authenticated (or, better said, when the authorization
  // header is not set, the fastLoginMiddleware, in the dev-server, skips
  // any authentication check)
  const cgnDeactivationEndpoint = "http://127.0.0.1:3000/api/v1/cgn/delete";
  // eslint-disable-next-line functional/no-let
  let responseCode = -1;
  try {
    const response = await fetch(cgnDeactivationEndpoint, {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      }
    });
    responseCode = response.status;
  } catch (error) {
    // We are not really interested about the error
    // eslint-disable-next-line no-console
    console.log(`deactivateCGNCardIfNeeded: ${error}`);
  }

  if (responseCode !== 201 && responseCode !== 403) {
    throw new Error(
      `Invalid response code from CGN deactivateion (${responseCode}), check the CGN deactivation endpoint (${cgnDeactivationEndpoint}) and the dev-server configuration for CGN deactivation`
    );
  }
};
