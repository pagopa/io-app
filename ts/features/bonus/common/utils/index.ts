import { bpdEnabled, cdcEnabled } from "../../../../config";

export const ID_BPD_TYPE = 2;
export const ID_CGN_TYPE = 3;
export const ID_CDC_TYPE = 4;

/**
 * Return a Map with the bonus ID as a key and
 * a bool representing the relative feature flag
 * status as a value.
 */
export const mapBonusIdFeatureFlag = () =>
  new Map<number, boolean>([
    [ID_BPD_TYPE, bpdEnabled],
    [ID_CGN_TYPE, true],
    [ID_CDC_TYPE, cdcEnabled]
  ]);
