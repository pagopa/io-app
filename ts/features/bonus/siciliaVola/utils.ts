import I18n from "../../../i18n";
import { AeroportiAmmessiInputBean } from "../../../../definitions/api_sicilia_vola/AeroportiAmmessiInputBean";
import { SvVoucher } from "./types/SvVoucher";
import {
  DisabledVoucherRequest,
  PartialVoucherRequest,
  SickVoucherRequest,
  StudentVoucherRequest,
  VoucherRequest,
  WorkerVoucherRequest
} from "./types/SvVoucherRequest";

export const fromVoucherToDestinationLabels = (voucher: SvVoucher) => {
  switch (voucher.category) {
    case "student":
      return [
        {
          label: I18n.t("bonus.sv.components.destinationLabels.universityName"),
          value: voucher.university.universityName
        },
        {
          label: I18n.t("bonus.sv.components.destinationLabels.municipality"),
          value: voucher.university.municipality.name
        }
      ];
    case "worker":
      return [
        {
          label: I18n.t("bonus.sv.components.destinationLabels.companyName"),
          value: voucher.company.businessName
        },
        {
          label: I18n.t("bonus.sv.components.destinationLabels.municipality"),
          value: voucher.company.municipality.name
        }
      ];
    case "sick":
      return [
        {
          label: I18n.t("bonus.sv.components.destinationLabels.hospitalName"),
          value: voucher.hospital.hospitalName
        },
        {
          label: I18n.t("bonus.sv.components.destinationLabels.municipality"),
          value: voucher.hospital.municipality.name
        }
      ];
    case "disabled":
      return [];
  }
};

const isStudentVoucherRequest = (
  partialVoucherRequest: PartialVoucherRequest
): partialVoucherRequest is StudentVoucherRequest =>
  partialVoucherRequest.category === "student" &&
  partialVoucherRequest.university !== undefined &&
  partialVoucherRequest.departureDate !== undefined;

const isWorkerVoucherRequest = (
  partialVoucherRequest: PartialVoucherRequest
): partialVoucherRequest is WorkerVoucherRequest =>
  partialVoucherRequest.category === "worker" &&
  partialVoucherRequest.underThresholdIncome !== undefined &&
  partialVoucherRequest.company !== undefined &&
  partialVoucherRequest.departureDate !== undefined;

const isSickVoucherRequest = (
  partialVoucherRequest: PartialVoucherRequest
): partialVoucherRequest is SickVoucherRequest =>
  partialVoucherRequest.category === "sick" &&
  partialVoucherRequest.underThresholdIncome !== undefined &&
  partialVoucherRequest.hospital !== undefined &&
  partialVoucherRequest.departureDate !== undefined;

const isDisabledVoucherRequest = (
  partialVoucherRequest: PartialVoucherRequest
): partialVoucherRequest is DisabledVoucherRequest =>
  partialVoucherRequest.category === "disabled" &&
  partialVoucherRequest.departureDate !== undefined;

export const isVoucherRequest = (
  partialVoucherRequest: PartialVoucherRequest
): partialVoucherRequest is VoucherRequest =>
  isStudentVoucherRequest(partialVoucherRequest) ||
  isWorkerVoucherRequest(partialVoucherRequest) ||
  isSickVoucherRequest(partialVoucherRequest) ||
  isDisabledVoucherRequest(partialVoucherRequest);

export const destinationsInfoFromVoucherRequest = (
  voucherRequest:
    | StudentVoucherRequest
    | WorkerVoucherRequest
    | SickVoucherRequest
): AeroportiAmmessiInputBean => {
  const destination =
    voucherRequest.category === "student"
      ? voucherRequest.university
      : voucherRequest.category === "worker"
      ? voucherRequest.company
      : voucherRequest.hospital;

  return {
    // TODO: check consistency between type in swagger.
    stato: destination.state.id.toString(),
    latitudine: destination.municipality.latitude,
    longitudine: destination.municipality.longitude
  };
};
