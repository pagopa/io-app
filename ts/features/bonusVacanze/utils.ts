/**
 * This is just a temporary file to define utility types to develop Bonus section
 */
export type Bonus = {
  type: string;
  code: string;
  max_amount: number;
  tax_benefit: number;
  activated_at: Date;
};
