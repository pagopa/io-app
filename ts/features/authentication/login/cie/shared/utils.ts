import { Platform } from "react-native";
import { isDevEnv } from "../../../../../utils/environment";
import { SpidLevel } from "../utils";

export type WebViewLoginNavigationProps = {
  spidLevel: SpidLevel;
  isUat: boolean;
};

export const iOSUserAgent =
  "Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1";

export const defaultUserAgent = Platform.select({
  ios: iOSUserAgent,
  default: undefined
});

export const originSchemasWhiteList = [
  "https://*",
  "iologin://*",
  ...(isDevEnv ? ["http://*"] : [])
];

export const WHITELISTED_DOMAINS = [
  "https://idserver.servizicie.interno.gov.it",
  "https://oidc.idserver.servizicie.interno.gov.it",
  "https://mtls.oidc.idserver.servizicie.interno.gov.it",
  "https://mtls.idserver.servizicie.interno.gov.it",
  "https://ios.idserver.servizicie.interno.gov.it",
  "https://ios.oidc.idserver.servizicie.interno.gov.it"
];

export type CieIdLoginProps = {
  spidLevel: SpidLevel;
  isUat: boolean;
};
