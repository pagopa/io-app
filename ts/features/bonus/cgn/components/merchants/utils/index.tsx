import { WithinRangeInteger } from "@pagopa/ts-commons/lib/numbers";
import * as E from "fp-ts/lib/Either";
import { pipe } from "fp-ts/lib/function";
import * as O from "fp-ts/lib/Option";
import { Discount } from "../../../../../../../definitions/cgn/merchants/Discount";
import I18n from "../../../../../../i18n";
import { getCategorySpecs } from "../../../utils/filters";

export const normalizedDiscountPercentage = (discount?: number) =>
  pipe(
    WithinRangeInteger(1, 100).decode(discount),
    E.map(v => v.toString()),
    E.getOrElse(() => "-")
  );

export const isValidDiscount = (discount?: number) =>
  pipe(WithinRangeInteger(1, 100).decode(discount), E.isRight);

export const moduleCGNaccessibilityLabel = (discountData: Discount) => {
  const { name, discount, productCategories, isNew } = discountData;

  const discountPercentage = isValidDiscount(discount)
    ? `${I18n.t("bonus.cgn.merchantsList.discountOf", {
        discount: normalizedDiscountPercentage(discount)
      })},`
    : "";

  const categories = `
    ${I18n.t("bonus.cgn.merchantsList.filter.categories")}:
    ${productCategories
      .map(categoryKey => {
        const category = getCategorySpecs(categoryKey);
        return O.isSome(category) ? I18n.t(category.value.nameKey) : "";
      })
      .join(", ")}
      `;

  const isNewLabel = isNew ? `${I18n.t("bonus.cgn.merchantsList.news")},` : "";

  return `${name}, ${isNewLabel} ${discountPercentage} ${categories}`;
};
