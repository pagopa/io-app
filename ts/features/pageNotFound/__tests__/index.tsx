import { fireEvent, render } from "@testing-library/react-native";
import {
  BTN_CLOSE_TEST_ID,
  BTN_UPDATE_TEST_ID,
  PageNotFound
} from "../screens";
import * as url from "../../../utils/url";

const mockPopToTop = jest.fn();
const mockOpenAppStoreUrl = jest.fn();

jest.mock("../../../navigation/params/AppParamsList", () => ({
  useIONavigation() {
    return {
      popToTop: mockPopToTop
    };
  }
}));

jest.spyOn(url, "openAppStoreUrl").mockImplementation(mockOpenAppStoreUrl);

describe(PageNotFound, () => {
  afterEach(jest.clearAllMocks);

  it("Should match the snapshot", () => {
    const component = render(<PageNotFound />);

    expect(component).toMatchSnapshot();
  });
  it("Should call openAppStoreUrl", () => {
    const { getByTestId } = render(<PageNotFound />);

    const btnUpdate = getByTestId(BTN_UPDATE_TEST_ID);
    fireEvent.press(btnUpdate);

    expect(mockOpenAppStoreUrl).toHaveBeenCalledTimes(1);
    expect(mockPopToTop).not.toHaveBeenCalled();
  });
  it("Should call popToTop", () => {
    const { getByTestId } = render(<PageNotFound />);

    const btnClose = getByTestId(BTN_CLOSE_TEST_ID);
    fireEvent.press(btnClose);

    expect(mockPopToTop).toHaveBeenCalledTimes(1);
    expect(mockOpenAppStoreUrl).not.toHaveBeenCalled();
  });
});
