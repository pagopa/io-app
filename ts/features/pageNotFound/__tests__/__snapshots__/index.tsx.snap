// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound Should match the snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flexGrow": 1,
      "marginHorizontal": 24,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      [
        {
          "alignContent": "center",
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
      ]
    }
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    waitFor={
      [
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={120}
          bbWidth={120}
          focusable={false}
          height={120}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 120,
                "width": 120,
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
          width={120}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGPath
              d="M102.76 55c27.83 0 50.39 22.56 50.39 50.39 0 1.8-.1 3.58-.28 5.33l-3.63-2.63-3.97-1.84s-5.58.69-6.49 0c-.91-.69-3.36.07-3.68.25-.32.18-3.22 3.07-3.22 3.07l-1.28 3.54v3.94l1.28 2.9v1.89l-5.56-1.38s-.81-.19-2.71-.63c-1.89-.44-7.77-.02-7.96-.87-.19-.85-.1-.92-1.53-1.1-1.43-.17-4.95-.81-4.95-.81l-2.77-.35-2.72-.24s-3.41.83-3.63.98c-.2299.15-2.6699 6.32-2.6699 6.32v1.98l-.93 1.54-2.44 1.36-3.35 3.73.43 8.12 4.02 4.73-4.02 5.58.99 3.82c-22.7-4.9-39.71-25.08-39.71-49.25 0-27.83 22.56-50.39 50.3899-50.39V55Z"
              fill={
                {
                  "payload": 4289392367,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M202.651 164.54c-.11-.13-10.98-13.21-23.59-26.68-20.84-22.26-30.19-28.43-34.36-29.69-5.3-1.6-9.25-.7-11.14 2.51-1.96 3.35-1.13 8.07.92 10.06.66.63.8 1.63.35 2.42-.45.79-1.38 1.18-2.26.94-.18-.05-18.54-4.97-29.01-5.63-1.05-.06-2.06.29-2.85.99-.8004.72-1.2704 1.72-1.3204 2.8-.09 2.07 1.3904 3.83 3.4304 4.11 9.84 1.33 22.46 3.45 26.75 6.03l-2.06 3.43c-2.84-1.71-12.03-3.71-25.22-5.49-4.1104-.55-7.0804-4.1-6.9004-8.25.1-2.17 1.04-4.16 2.65-5.6 1.59-1.43 3.6404-2.14 5.7704-2.01 7.67.48 18.97 3.05 25.23 4.59-1.02-3.37-.72-7.37 1.07-10.42 1.36-2.31 5.61-7.36 15.74-4.31 15.87 4.78 58.09 55.48 59.88 57.64l-3.08 2.55v.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M122.41 150.3c-2.3-1.73-10.9-2.99-22.9898-3.37-5.84-.18-10.47-5.09-10.32-10.94.11-4.47 3.04-8.44 7.28-9.87l2.11-.71 1.27 3.79-2.11.71c-2.66.89-4.49 3.37-4.55 6.17-.09 3.66 2.8 6.73 6.45 6.85 13.6498.43 22.1598 1.83 25.2698 4.17l-2.4 3.2h-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M162.49 209.47c-16.99-25.12-39-41.13-65.4198-47.6l-.51-.12c-2.85-.7-5.15-2.77-6.15-5.53-1.04-2.86-.5-6.12 1.4-8.5l3.29-4.1 3.12 2.5-3.29 4.1c-1.04 1.3-1.33 3.07-.76 4.64.55 1.51 1.8 2.63 3.35 3.02l.5.12c27.4098 6.71 50.2198 23.28 67.7798 49.25l-3.31 2.24v-.02ZM142.327 127.059l-2.585 3.052 8.021 6.792 2.585-3.052-8.021-6.792Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M150.6 155.74c-6.01-10.42-2.18-27.59-2.01-28.32l3.9.89c-.04.16-3.65 16.38 1.58 25.42l-3.46 2-.01.01ZM130.063 127.83l-12.575 27.227 3.632 1.677 12.574-27.226-3.631-1.678Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M123.15 84.47c-.39 0-.79-.15-1.08-.46-.57-.6-.55-1.55.05-2.12l31.26-29.92c.6-.57 1.55-.55 2.12.05.57.6.55 1.55-.05 2.12l-31.26 29.92c-.29.28-.66.42-1.04.42v-.01ZM106.62 75.54c-.82 0-1.49-.66-1.5-1.48l-.45-41.54c0-.83.66-1.51 1.48-1.52h.02c.82 0 1.49.66 1.5 1.48l.45 41.54c0 .83-.66 1.51-1.48 1.52h-.02ZM87.8601 82.24c-.39 0-.78-.15-1.07-.45l-26.8-27.47c-.58-.59-.57-1.54.03-2.12.59-.58 1.54-.57 2.12.03l26.8 27.47c.58.59.57 1.54-.03 2.12-.29.28-.67.43-1.05.43v-.01ZM82.0598 100.99c-.06 0-.12 0-.18-.01l-45.55-5.36c-.82-.1-1.41-.84-1.31-1.67.1-.82.84-1.41 1.66-1.31l45.56 5.36c.82.1 1.41.84 1.31 1.67-.09.76-.74 1.32-1.49 1.32Z"
              fill={
                {
                  "payload": 4278240714,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "height": 24,
            }
          }
        />
      </View>
      <Text
        accessibilityRole="header"
        allowFontScaling={true}
        dynamicTypeRamp="title2"
        maxFontSizeMultiplier={1.5}
        style={
          [
            {},
            {
              "color": "#0E0F13",
              "fontFamily": "Titillio",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "600",
              "lineHeight": 33,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Nothing here!
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <Text
        allowFontScaling={true}
        dynamicTypeRamp="body"
        maxFontSizeMultiplier={1.5}
        style={
          [
            {},
            {
              "color": "#555C70",
              "fontFamily": "Titillio",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        This page does not exist or is not available in the version of the app you are using.
Try updating IO.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Update the IO app"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "auto",
                "flexShrink": 1,
              }
            }
            testID="pageNotFoundBtnUpdateId"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "borderCurve": "continuous",
                    "borderRadius": 8,
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingHorizontal": 24,
                    "textAlignVertical": "center",
                  },
                  {
                    "height": 48,
                  },
                  {
                    "backgroundColor": "#0B3EE3",
                    "overflow": "hidden",
                  },
                  false,
                  {},
                  false,
                  {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "columnGap": 8,
                    },
                    false,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  accessible={false}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  importantForAccessibility="no-hide-descendants"
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    [
                      {},
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "Titillio",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "600",
                        "lineHeight": 20,
                      },
                      {
                        "alignSelf": "center",
                      },
                    ]
                  }
                >
                  Update the IO app
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Close"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 14,
                "left": 24,
                "right": 24,
                "top": 14,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTouchEnd={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            testID="pageNotFoundBtnCloseId"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "textAlignVertical": "center",
                  },
                  false,
                  {
                    "columnGap": 8,
                  },
                  {},
                  {
                    "columnGap": 8,
                  },
                  false,
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                accessible={false}
                allowFontScaling={true}
                ellipsizeMode="tail"
                importantForAccessibility="no-hide-descendants"
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  [
                    {},
                    {
                      "color": "#0E0F13",
                      "fontFamily": "Titillio",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "600",
                      "lineHeight": 20,
                    },
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "textAlign": "auto",
                      },
                    ],
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
